### PR TITLE
Break out main and create a core sub module for running

### LINKED ISSUE #69: Test command line argument processing
Just had an issue reported that would have hopefully been caught with testing of the parameters and how they are processed. 

Add testing so we can make sure we are honoring the parameters correctly.

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:* 
Fixes #69
#47 

*Description of changes:*
- Break out __main__ logic into a new sub module core
- Write tests around the new core functions
- Rename Parser to cfn_yaml

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by fatbasstard** [comment]
Also helps for https://github.com/awslabs/cfn-python-lint/issues/47

File: src/cfnlint/core.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+            if config_name == 'log_level':
+                if isinstance(config_value, (six.string_types, six.text_type)):
+                    defaults['log_level'] = config_value
+
+    return defaults
+
+
+def run_checks(filename, template, rules, transforms, ignore_checks, regions):
+    """Run Checks against the template"""
+    if regions:
```
Comment (cmmeyer) [inline src/cfnlint/core.py]: Does it make sense to break out supported regions into a constant?  That way we can re-use elsewhere and easily update if more regions are added.

Comment (kddejong) [reply src/cfnlint/core.py]: Most recent change should cover this.  We were also setting a Regions variable in helpers and so I removed this one and used that one.

Comment (cmmeyer) [reply src/cfnlint/core.py]: Ha! No wonder I had the idea to break them out in my head -- we already had! :)  Looks good.

👍

**Pull request merged**